src/Simplex/Messaging/Notifications/Server.hs

```diff
@@ -154,11 +160,15 @@ ntfSubscriber NtfSubscriber {smpSubscribers, newSubQ, smpAgent = ca@SMPClientAge
   where
     subscribe :: M ()
     subscribe =
-      forever $
-        atomically (readTBQueue newSubQ) >>= \case
-          sub@(NtfSub NtfSubData {smpQueue = SMPQueueNtf {smpServer}}) -> do
-            SMPSubscriber {newSubQ = subscriberSubQ} <- getSMPSubscriber smpServer
-            atomically $ writeTQueue subscriberSubQ sub
+      forever $ do
+        subs <- atomically (readTBQueue newSubQ)
+        let ss = L.groupBy ((==) `on` server) subs
```

Comment:
why groupBy is needed both here and in `resubscribe` - is it a sanity check?

Reply:
they operate on slightly different types, the top one on `NonEmpty NtfSubData`, the bottom one on `NonEmpty (NtfEntityRec 'Subscription)`, where the latter type is a wrapper around `NtfSubData`

Also they are called in different cases - the first when the server just starts, and the latter when ntfSubscriber receives the new subscriptions...

But maybe the latter is indeed sufficient and the former can just dump there the whole array without splitting by server? Need to think.

Reply:
The only difference is the delay - initially it was after each subscription, not sure if it was helpful or not, now it's between groups of subscriptions on different servers, and leaving only two would remove the delay completely...

Reply:
I don't remember why this delay was added in the first place?